external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaContinuousReader.scala

```diff
@@ -53,7 +54,7 @@ class KafkaContinuousReader(
     metadataPath: String,
     initialOffsets: KafkaOffsetRangeLimit,
     failOnDataLoss: Boolean)
-  extends ContinuousReader with SupportsScanUnsafeRow with Logging {
```

Comment:
can we get rid of `KafkaRecordToUnsafeRowConverter`? Since Spark would do a unsafe projection at the end, here we should just return `GenericInternalRow` instead of `UnsafeRow`, to save data cpoy.

Reply:
We can, but this is intended to make minimal changes. We can add optimizations like this in a follow-up.